### DISCUSSION TITLE
Change port inside .env

### DISCUSSION THREAD

**Discussion by zakcodez** [vercel/next.js#42875] [Help]
How do you change the port **without** baking it into the npm script, but putting it into `.env` ?

**Comment by icyJoseph** [comment]
Mmm I think the PORT is loaded from the environment before the `.env` file is applied to the app build/run context.

If we try to do:

```json
  "scripts": {
    "dev": "next dev",
    "dev:port": "export $(cat .env | xargs) && next dev --port $PORT",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
  },
```

Shell scripting is not my strongest but this should do the trick for most Linux/MacOS users, not sure about Windows.
